File: 프로그래머스/unrated/181839. 주사위 게임 1/주사위 게임 1.java
class Solution {
    public int solution(int a, int b) {
       if(a % 2 == 1 && b % 2 == 1){
           return (int)Math.pow(a,2) + (int)Math.pow(b,2);
       }
        if(a % 2 == 1 || b % 2 == 1){
            return 2*(a + b);
        }
        return (int)Math.abs(a-b);
    }
}

File: 프로그래머스/0/181839. 주사위 게임 1/주사위 게임 1.cpp
#include <string>
#include <vector>
#include <cmath>

using namespace std;

int solution(int a, int b) {
    return (a % 2 == 0 && b %  2 == 0) ? abs(a-b) : (a % 2 == 1 && b % 2 == 1) ? pow(a, 2) + pow(b, 2) : 2 *(a + b);
}